webapp-django/crashstats/signature/views.py

```diff
@@ -502,22 +502,30 @@ def signature_summary(request, params):
 
     # Augment graphics adapter with data from another service.
     if 'adapter_vendor_id' in facets:
-        vendor_hexes = set()
-        adapter_hexes = set()
+        vendor_hexes = []
+        adapter_hexes = []
         for vendor in facets['adapter_vendor_id']:
-            vendor_hexes.add(vendor['term'])
             for adapter in vendor['facets']['adapter_device_id']:
-                adapter_hexes.add(adapter['term'])
+                vendor_hexes.append(vendor['term'])
+                adapter_hexes.append(adapter['term'])
+
         devices_api = models.GraphicsDevices()
-        try:
-            devices_results = devices_api.get(
-                vendor_hex=vendor_hexes,
-                adapter_hex=adapter_hexes,
+
+        devices_results = {
+            'hits': [],
+            'total': 0,
+        }
+        # In order to avoid hitting the maximum URL size limit, we split the
+        # query in smaller chunks, and then we reconstruct the results.
+        for i in range(0, len(vendor_hexes), 50):
```

Comment:
I think getting the right number here is going to be the most important part. I got to ``50`` by just guessing from what works on stage and what doesn't, but there's no science there. 

Also as we discovered it's not nginx that blocks but uwsgi, so we are capped at 4096 bytes according to the logs you shared earlier. Can you find out much the "empty" query weights?